### ISSUE TITLE
Change online extension URL from `moz-extension.kiwix.org` to `extension.kiwix.org`?

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#1017]
@kelson42 Since a major piece of work migrating the Chromium extension to Manifest V3, is pretty much ready to merge (#984), I was wondering whether we should now change the online extension URL from `moz-extension.kiwix.org` to `extension.kiwix.org`, since both Firefox AND Chromium will now use this online extension code for Service Worker mode.

Of course, I can use `moz-extension...` also for the Chromium extension, but it might be a bit confusing to users. What do you think? Or could we consider it a bit of a plug for Mozilla in the face of Chrome's dominance? 😉

If we do migrate the (sub)-domain name, I would need to be able to have a transparent redirect from `moz-extension...` to `extension...` so that users of older versions of the Firefox extension (which have `moz-extension` set as the online address) can still reach the extension. In other words, I'd need to have control over both domains for some months, possible a year or so.

**Comment by kelson42** [comment]
@Jaifroid Doable, but "browser-extension.kiwix.org" might be better no?

**Comment by Jaifroid** [comment]
Yes, good idea. This is what local URLs look like in the various browsers:

* Firefox: `moz-extension://32fa3a6c-575b-4d18-a833-7a7a500e762f/www/index.html`
* Chrome: `chrome-extension://donaljnlmapmngakoipdmehbfcioahhk/www/index.html`
* Edge: `extension://fbbdjelfpndiajmbaocbfcjdgomfbjgm/www/index.html`

I guess we want something that won't look too jarringly different for any user. Either:

* `https://browser-extension.kiwix.org/current/www/index.html`; or
* `https://extension.kiwix.org/current/www/index.html`

I agree that `browser-extension` looks good and is neutral. How do I go about getting that subdomain provisioned? I'd update it using a container exactly like the current moz-extension subdomain.

Docker files are here: https://github.com/kiwix/kiwix-js/tree/main/docker, and the workflow that updates it is: https://github.com/kiwix/kiwix-js/blob/main/.github/workflows/publish-extension.yaml.

**Comment by kelson42** [comment]
@Jaifroid I can change the dns today so we have the new fqdn. Old one woukd be a permanent redirect to new one.

**Comment by Jaifroid** [comment]
Hmm, I think this would have to be co-ordinated with the release of the new version, or else the old ones could stop working, as some of their functionality involves receiving messages from a specified domain... I'll need to check...

**Comment by Jaifroid** [comment]
I might need to run both domains for a while, until I'm sure everything is working fine. Would this be possible?

**Comment by Jaifroid** [comment]
In any case, I'll confirm once I've double-checked the code. Would the scripts for updating the container still work? If they operate via an IP address as opposed to a specified domain name, I guess they should. (I'm unsure, because updating GHCR is done by rgaudin's scripts and the keys/secrets he provided).

**Comment by kelson42** [comment]
@rgaudin @Jaifroid Can you please secure the container can deliver for both domain names? Then I would create a new DNS entry pointing to the same container.

**Comment by rgaudin** [comment]
I believe the code is not host-bound and I've done my part: https://extension.kiwix.org

**Comment by Jaifroid** [comment]
Thank you @rgaudin! That is working. So if I update moz-extension.kiwix.org it will also update extension.kiwix.org because both names point to the same server? And presumably I can migrate the container / GHCR workflow to updating extension.kiwix.org directly? Sorry for probably basic questions!

**Comment by rgaudin** [comment]
Currently the two domain names points to the same server and the server config points those two domains to the same container.

Please restate exactly what you want to I can update the config (I don't quite follow the discussion).

Currently:

- `moz-extension.kiwix.org` ➡️ `moz-extension-service`
- `extension.kiwix.org` ➡️ `moz-extension-service`

**Comment by rgaudin** [comment]
Sorry after talking to @kelson42 I changed it to:

- `moz-extension.kiwix.org` ➡️ `moz-extension-service`
- `browser-extension.kiwix.org` ➡️ `moz-extension-service`

https://browser-extension.kiwix.org/ is now the new one, https://moz-extension.kiwix.org/ still works until @Jaifroid decides to shut it down and `extension.kiwix.org` has been removed.

**Comment by kelson42** [comment]
@Jaifroid Ball back on your side if I understand properly :)

**Comment by Jaifroid** [comment]
Wonderful, thank you both! I'll let you know if/when moz-extension can be removed.
